homeassistant/components/mychevy.py

```diff
@@ -0,0 +1,121 @@
+from logging import getLogger
+from datetime import timedelta
+import time
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_USERNAME, CONF_PASSWORD)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers import discovery
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ["mychevy==0.1.1"]
+
+DOMAIN = 'mychevy'
+
+MYCHEVY_SUCCESS = "Success"
+MYCHEVY_ERROR = "Error"
+
+NOTIFICATION_ID = 'mychevy_website_notification'
+NOTIFICATION_TITLE = 'MyChevy website status'
+
+_LOGGER = getLogger(__name__)
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=30)
+ERROR_SLEEP_TIME = timedelta(minutes=30)
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+class EVSensorConfig(object):
+    """Configuration for EV Sensors."""
+
+    def __init__(self, name, attr, unit_of_measurement=None,
+                 icon=None, icon_func=None):
+        """Init EV Sensors."""
+        self.name = name
+        self.attr = attr
+        self.unit_of_measurement = unit_of_measurement
+        self.icon = icon
+        self.icon_func = icon_func
+
+
+class EVBinarySensorConfig(object):
+    """Config for binary sensors."""
+
+    def __init__(self, name, attr, device_class=None):
+        """Init config."""
+        self.name = name
+        self.attr = attr
+        self.device_class = device_class
+
+
+@asyncio.coroutine
+def async_setup(hass, base_config):
+    """Setup mychevy platform."""
+    import mychevy.mychevy as mc
+
+    config = base_config.get(DOMAIN)
+    _LOGGER.debug('Received configuration: %s', config)
+
+    email = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    if hass.data.get(DOMAIN) is None:
+        hass.data[DOMAIN] = MyChevyHub(mc.MyChevy(email, password), hass)
+
+    discovery.load_platform(hass, 'sensor', DOMAIN, {}, config)
+    discovery.load_platform(hass, 'binary_sensor', DOMAIN, {}, config)
+    hass.async_add_job(hass.data[DOMAIN].update)
+
+    return True
+
+
+class MyChevyHub(object):
+    """MyChevy Hub.
+
+    Connecting to the mychevy website is done through a selenium
+    webscraping process. That can only run synchronously. That call
+    takes 2 - 3 minutes to complete, so must only be done with
+    async_add_job.
+    """
+
+    def __init__(self, client, hass, **kwags):
+        """Initialize MyChevy Hub."""
+        super(MyChevyHub, self).__init__()
+        self._client = client
+        self.hass = hass
+        self.car = None
+        self.status = None
+        self.sensors = []
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self):
+        """Update call that is safe to call from async_add_job."""
+        try:
+            self.update()
+            self.car = self._client.data()
+            self.status.success()
+        except Exception:
+            _LOGGER.exception(
```

Comment:
We usually don't want to log with exception level in home assistant, error level is preferred.